File: src/main/java/no/nav/sosialhjelp/soknad/business/db/soknadmetadata/BatchSoknadMetadataRepository.java
package no.nav.sosialhjelp.soknad.business.db.soknadmetadata;


import no.nav.sosialhjelp.soknad.business.domain.SoknadMetadata;

import java.util.Optional;

public interface BatchSoknadMetadataRepository {

    Optional<SoknadMetadata> hentForBatch(int antallDagerGammel);
    Optional<SoknadMetadata> hentEldreEnn(int antallDagerGammel);
    void leggTilbakeBatch(Long id);
    void slettSoknadMetaData(String behandlingsId, String eier);
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/soknadmetadata/BatchSoknadMetadataRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.soknadmetadata

interface BatchSoknadMetadataRepository {
    fun hentForBatch(antallDagerGammel: Int): SoknadMetadata?
    fun hentEldreEnn(antallDagerGammel: Int): List<SoknadMetadata>
    fun leggTilbakeBatch(id: Long)
    fun slettSoknadMetaDataer(behandlingsIdList: List<String>)
}
